org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java

```diff
@@ -289,6 +300,21 @@ public boolean isCanceled() {
 				}
 			}
 		}
+		List<CompletionItem> tempProposals = proposals.stream().filter(prop -> prop.getKind() == CompletionItemKind.Keyword || prop.getKind() == CompletionItemKind.Snippet).collect(Collectors.toList());
+		tempProposals.sort(LABEL_COMPARATOR);
+		for (int i = 0; i < tempProposals.size() - 1; i++) {
+			CompletionItem currentItem = tempProposals.get(i);
+			CompletionItem nextItem = tempProposals.get(i + 1);
+			if (currentItem.getLabel().equals(nextItem.getLabel())) {
+				String newSortText;
+				if (currentItem.getKind() == CompletionItemKind.Keyword) {
+					newSortText = Integer.toString(Integer.parseInt(currentItem.getSortText()) - 1);
+				} else {
+					newSortText = Integer.toString(Integer.parseInt(nextItem.getSortText()) - 1);
```

Comment:
Yes, that's what it does. It don't think it ever gets triggered in practice because the keywords always precede the snippets initially in the sorting, but I figured it was safer to include it just in case.

Reply:
Ok. The thing that made me think hard was `nextItem.getSortText()`. If it's truly a keyword (which logically makes sense) then we're fine. But if there is a case where 2 snippets can have the same label, and given that our generic snippets don't have `sortText` set normally, then this might cause problems. That's pretty obscure though so I really think it should be ok.